internal/emma/validation/kubernetes_resource_validation.go

```diff
@@ -0,0 +1,300 @@
+package emma
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"regexp"
+	"slices"
+)
+
+type KubernetesResourceName struct{}
+
+func (k KubernetesResourceName) Description(ctx context.Context) string {
+	return "The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between."
+}
+
+func (k KubernetesResourceName) MarkdownDescription(ctx context.Context) string {
+	return "The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between."
+}
+
+func (k KubernetesResourceName) ValidateString(ctx context.Context, req validator.StringRequest, resp *validator.StringResponse) {
+	if req.ConfigValue.IsUnknown() || req.ConfigValue.IsNull() {
+		return
+	}
+	matches, _ := regexp.MatchString(`^([a-z](?:[0-9a-z-]{0,62}[0-9a-z]))$`, req.ConfigValue.ValueString())
+	if !matches {
+		resp.Diagnostics.AddError("Validation Error", "Action rejected: The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between.")
```

Comment:
Will user understand what name must be valid?

Reply:
Just copied from doc, fixed thanks